src/chainstate/stacks/db/blocks.rs

```diff
@@ -0,0 +1,3459 @@
+        let args = [&burn_hash.to_hex() as &dyn ToSql, &anchored_block_hash.to_hex() as &dyn ToSql, &last_seq as &dyn ToSql];
+        let mut staging_microblocks = query_rows::<StagingMicroblock, _>(tx, &sql, &args).map_err(Error::DBError)?;
+
+        // load associated staging microblock data, if present 
+        for i in 0..staging_microblocks.len() {
+            staging_microblocks[i].block_data = StacksChainState::load_staging_microblock_bytes(tx, &staging_microblocks[i].microblock_hash)?.unwrap_or(vec![]);
+        }
+
+        // what's the first microblock in this stream?
+        let first_microblock_hash = match StacksChainState::get_microblock_stream_head_hash(tx, burn_hash, anchored_block_hash)? {
+            Some(bhh) => bhh,
+            None => {
+                unreachable!("BUG: No staging microblocks head hash, but loaded staging microblocks!");
+            }
+        };
+
+        // merge with chunk-stored microblock stream, if present
+        let stored_microblocks = match StacksChainState::load_microblock_stream(tx.get_blocks_path(), burn_hash, &first_microblock_hash).or_else(StacksChainState::empty_stream)? {
+            Some(mblocks) => mblocks,
+            None => vec![]
+        };
+    
+        let microblocks = StacksChainState::merge_microblock_streams(staging_microblocks, stored_microblocks)?;
+
+        debug!("Accept microblock stream rooted at {}/{}-{} (up to {})", burn_hash.to_hex(), anchored_block_hash.to_hex(), microblocks[0].block_hash().to_hex(), last_seq);
+        StacksChainState::store_microblock_stream(tx.get_blocks_path(), burn_hash, &microblocks)?;
+
+        // clear out of staging
+        let sql = "UPDATE staging_microblocks SET processed = 1 WHERE burn_header_hash = ?1 AND anchored_block_hash = ?2 AND sequence <= ?3".to_string();
+        let args = [&burn_hash.to_hex() as &dyn ToSql, &anchored_block_hash.to_hex() as &dyn ToSql, &last_seq as &dyn ToSql];
+
+        tx.execute(&sql, &args)
+            .map_err(|e| Error::DBError(db_error::SqliteError(e)))?;
+
+        for i in 0..microblocks.len() {
+            StacksChainState::delete_staging_microblock_data(tx, &microblocks[i].block_hash())?;
+        }
+
+        Ok(())
+    }
+     
+    /// Given a microblock stream, does it connect the parent and child anchored blocks?
+    /// * verify that the blocks are a contiguous sequence, with no duplicate sequence numbers
+    /// * verify that each microblock is signed by the parent anchor block's key
+    /// The stream must be in order by sequence number, and there must be no duplicates.
+    /// If the stream connects to the anchored block, then
+    /// return the index in the given microblocks vec that corresponds to the highest valid
+    /// block -- i.e. the microblock indicated by the anchored header as the parent.
+    /// If there was a duplicate sequence number, then also return a poison-microblock
+    /// transacti
```

Comment:
Should we use `BurnDB::get_last_snapshot_with_sortition` instead of ` BurnDB::get_block_snapshot_in_fork`?
ie can we end up in a situation where the block at height (block_height -1) does not include a sortition but the block at height (block_height - 2) does?

Reply:
I'm not sure where you're referring to?  This method doesn't call either of those.  Do you mean in `validate_anchored_block_burnchain()`?

Reply:
Oh that is weird bug from github...

<img width="884" alt="Screen Shot 2020-01-03 at 10 23 48 AM" src="https://user-images.githubusercontent.com/87777/71731638-41b72780-2e13-11ea-881c-9c16bd3d0c24.png">

Reply:
Ah, thanks.  The reason we use `get_last_snapshot_in_fork` instead of `get_last_snapshot_with_sortition` is because we're interested in getting the latest `sortition_hash` field, even if no sortition happened in the last Bitcoin block.  Miners always prove on the latest `sortition_hash`, and the sortition code ensures that it will be both unique to each Bitcoin block (even if there is no sortition) and unpredictable calculating it from the hash of the last Bitcoin block's sortition hash, the winning VRF seed (if a sortition happened), and the new Bitcoin block header.